kmip/pie/objects.py

```diff
@@ -227,6 +235,7 @@ def __init__(self, algorithm, length, value, masks=None,
         self.value = value
```

Comment:
It's generally not recommended to focus heavily on type checking in Python. As you are aware, Python relies heavily on [duck typing](https://en.wikipedia.org/wiki/Duck_typing#In_Python), where it's the interface of an object that matters, not its actual type. The onus is generally placed on good documentation, clear code, sound testing, and robust exception handling to make sure that code and data are used properly.

Given how everything is public and touchable in Python, type checking doesn't really guarantee much. A type check only verifies that things are valid at a specific point in time, at a specific point of execution. Anything before or after is vulnerable to modification.

I usually tend to shy away from lots of type checking, though for PyKMIP the pattern I've used is to define `validate` methods that check input data upon creation of new objects, like keys. In this case, I already check that `value` is a `bytes` object, down in the `validate` method below.